devito/types/dimension.py

```diff
@@ -1060,6 +1064,36 @@ def __new__(cls, d, name):
     _pickle_kwargs = []
 
 
+class CustomDimension(BasicDimension):
+
+    """
+    Dimension defining an iteration space with custom (known) size.
+
+    Notes
+    -----
+    This could be extended in the future for more customization (e.g., min point,
+    max point, etc.).
+    """
+
+    is_Custom = True
+
+    def __init_finalize__(self, name, symbolic_size=None):
+        self._symbolic_size = symbolic_size
+
+    @property
+    def symbolic_size(self):
+        return self._symbolic_size
+
+    def _arg_defaults(self):
+        return {}
+
+    def _arg_values(self, *args, **kwargs):
+        return {}
+
+    # Pickling support
+    _pickle_kwargs = ['symbolic_size']
```

Comment:
`BasicDimension._pickle_kwargs +  ['symbolic_size'] ` ?

Reply:
no, only `symbolic_size` needs to be pickled (there's no "spacing", like instead in the superclass, here)

also added a pickling test